src/indicator_cumulative_number.Rmd

```diff
@@ -65,7 +70,131 @@ In order to not consider subspecies, please take care of grouping by `species` (
 
 About point 7 of [#21](https://github.com/trias-project/pipeline/issues/21), please see this [comment](https://github.com/trias-project/pipeline/issues/21#issuecomment-381164874).
 
-## Cumulative number of alien species
+## Number of alien species
 
 see GitHub issue [#20](https://github.com/trias-project/pipeline/issues/20)
 
+```{r determine_presence_per_specieskey_per_year}
+#' Create total invasive species indicator plot (Trias)
+#'
+#' @param df data.frame Contains the data as produced by the Trias pipeline,
+#' with minimal colums.
+#' @param start_year_plot int Limit to use as start year of the plot. For scientific
+#' usage, the entire period could be relevant, but for policy purpose, focusing
+#' on a more recent period could be required.
+#' @param x_major_scale_stepsize int On which year interval labels are placed 
+#' on the x axis.
+#' @param x_minor_scale_stepsize int On which year interval minor breaks are
+#' placed on the x axis.
+#' @param facet_column NULL | char The column to use to create additional
+#' facet wrap plots underneath the main graph. When NULL, no facet graph is
+#' included.
+#'
+#' @return ggplot2 object
+#' @export
+#' @importFrom assertthat assert_that
+#' @importFrom assertable assert_colnames
+#' @importFrom dplyr distinct_ %>% filter rowwise do bind_cols group_by_ 
+#' count ungroup
+#' @importFrom tidyr replace_na unnest
+#' @importFrom lubridate year now
+#' @importFrom ggplot2 geom_line aes xlab ylab scale_x_continuous facet_wrap
+#' @importFrom INBOtheme theme_inbo
+#' @importFrom egg ggarrange
+#'
+#' @examples
+indicator_total_year <- function(df, start_year_plot = 1940, 
+                                 x_major_scale_stepsize = 10,
+                                 x_minor_scale_stepsize = 5,
+                                 facet_column = NULL) {
+  
+  # initial input checks
+  assert_that(is.data.frame(df))
+  assert_that(x_major_scale_stepsize >= x_minor_scale_stepsize)
+  assert_colnames(df, c("key", "first_observed", "last_observed"), 
+                  only_colnames = FALSE)
+  
+  if (is.null(facet_column)) {
+    df_cleaned <- df %>%
+      distinct_("speciesKey", "first_observed")
+  } else {
+    # check for valid facet options
+    valid_facet_options <- c("family", "order", "class", "phylum", 
```

Comment:
- do we really need to keep using the `native range` instead of `native_range`?
- if these facet_options need to be used elsewhere for other functions as well, keep it out of the function and define the vector of valid options as a package wide variable.

Reply:
You are right. I am planning to do it at `trias´ package level.

Reply:
About `native_range`/ `native_range`: you are right, we should underscore and *clean* column variables as much as possible. I remember I couldn't use `janitor` because `clean_names()` changes important GBIF related columns names. See example here below:
```r
a <- data.frame(scientificName= c("Homo informaticus"))
a
     scientificName
1 Homo informaticus

library(janitor)
clean_names(a)
     scientificname
1 Homo informaticus
```
The same holds true with `speciesKey` and many others columns. 
@stijnvanhoey : I don't find any other way than changing it manually with `dplyr::rename()`. I will do it in the pipeline for unifying checklists.